src/ddscxx/src/org/eclipse/cyclonedds/pub/qos/DataWriterQosDelegate.cpp

```diff
@@ -243,7 +253,8 @@ DataWriterQosDelegate::operator ==(const DataWriterQosDelegate& other) const
 #ifdef  OMG_DDS_OWNERSHIP_SUPPORT
            other.strength_    == strength_    &&
 #endif
-           other.lifecycle_   == lifecycle_;
+           other.lifecycle_   == lifecycle_   &&
+           other.datarepresentation_ == datarepresentation_;
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro.

Reply:
Will do.